zrml/prediction-markets/src/migrations.rs

```diff
@@ -157,7 +167,141 @@ impl<T: Config> OnRuntimeUpgrade for MigrateMarketPoolsBeforeOpen<T> {
+        Ok(())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() -> Result<(), &'static str> {
+        let dispute_period = T::DisputePeriod::get();
+        let current_block: T::BlockNumber = <frame_system::Pallet<T>>::block_number();
+        let last_dp_end_block = current_block.saturating_sub(dispute_period);
+        type DisputeBlockToMarketIdsTuple<T> =
+            (<T as frame_system::Config>::BlockNumber, BoundedVec<MarketIdOf<T>, CacheSize>);
+        type IterType<T> = PrefixIterator<DisputeBlockToMarketIdsTuple<T>>;
+
+        let mut market_ids_per_dispute_iterator: IterType<T> =
+            frame_support::migration::storage_key_iter::<_, _, Twox64Concat>(
+                b"PredictionMarkets",
+                b"MarketIdsPerDisputeBlock",
+            );
+        market_ids_per_dispute_iterator.try_for_each(
+            |(dispute_start_block, market_ids)| -> Result<(), &'static str> {
+                assert!(
+                    dispute_start_block > last_dp_end_block,
+                    "found unexpected storage key in MarketIdsPerDisputeBlock. \
+                     dispute_start_block: {:?}, last_dp_end_block: {:?}",
+                    dispute_start_block,
+                    last_dp_end_block
+                );
+
+                market_ids.iter().try_for_each(|market_id| -> Result<(), &'static str> {
+                    let market = T::MarketCommons::market(market_id)
+                        .map_err(|_| "invalid market_id found.")?;
+                    assert!(
+                        market.status == MarketStatus::Disputed,
+                        "found unexpected market status. market_id: {:?}, status: {:?}",
+                        market_id,
+                        market.status
+                    );
+                    Ok(())
+                })?;
+                Ok(())
+            },
+        )?;
+        let mut market_ids_per_reported_iterator: IterType<T> =
+            frame_support::migration::storage_key_iter::<_, _, Twox64Concat>(
+                b"PredictionMarkets",
+                b"MarketIdsPerReportBlock",
+            );
+        market_ids_per_reported_iterator.try_for_each(
+            |(dispute_start_block, market_ids)| -> Result<(), &'static str> {
+                assert!(
+                    dispute_start_block > last_dp_end_block,
+                    "found unexpected storage key in MarketIdsPerReportBlock. \
+                     dispute_start_block: {:?}, last_dp_end_block: {:?}",
+                    dispute_start_block,
+                    last_dp_end_block
+                );
 
+                market_ids.iter().try_for_each(|market_id| -> Result<(), &'static str> {
+                    let market = T::MarketCommons::market(market_id)
+                        .map_err(|_| "invalid market_id found.")?;
+                    assert!(
+                        market.status == MarketStatus::Reported,
```

Comment:
I think this can also be `Disputed`. But I am not sure. @maltekliemann @sea212 ?